src/porepy/grids/gmsh/gmsh_interface.py

```diff
@@ -92,79 +92,113 @@ def __write_fractures_compartments_2d(self):
         # Both fractures and compartments are
         constants = gridding_constants.GmshConstants()
 
-        frac_ind = np.argwhere(
-            np.logical_or(
+        # We consider fractures, boundary tag, an auxiliary tag (fake fractures/mesh constraints)
+        ind = np.argwhere(
+            np.logical_or.reduce((
                 self.lines[2] == constants.COMPARTMENT_BOUNDARY_TAG,
                 self.lines[2] == constants.FRACTURE_TAG,
-            )
+                self.lines[2] == constants.AUXILIARY_TAG
+            ))
         ).ravel()
-        frac_lines = self.lines[:, frac_ind]
+        lines = self.lines[:, ind]
+        tag = self.lines[2, ind]
 
-        frac_id = frac_lines[3, :]
-        if frac_id.size == 0:
+        lines_id = lines[3, :]
+        if lines_id.size == 0:
             return str()
-        range_id = np.arange(np.amin(frac_id), np.amax(frac_id) + 1)
+        range_id = np.arange(np.amin(lines_id), np.amax(lines_id) + 1)
 
-        s = "// Start specification of fractures\n"
+        s = "// Start specification of fractures/compartment boundary/auxiliary elements\n"
         seg_id = 0
         for i in range_id:
             local_seg_id = str()
-            for mask in np.flatnonzero(frac_id == i):
+            for mask in np.flatnonzero(lines_id == i):
+
+                # give different name for fractures/boundary and auxiliary
+                if tag[mask] != constants.AUXILIARY_TAG:
+                    name = "frac_line_"
+                    physical_name = constants.PHYSICAL_NAME_FRACTURES
+                else:
+                    name = "seg_line_"
+                    physical_name = constants.PHYSICAL_NAME_AUXILIARY
+
                 s += (
-                    "frac_line_"
+                    name
```

Comment:
I am not sure I agree with black's splitting of this one, but I suppose we're not entitled to such opinions.